src/analysis/special_functions/gamma/bohr_mollerup.lean

```diff
@@ -377,4 +379,91 @@ end
 
 end strict_mono
 
+section doubling
+
+/-!
+## The Gamma doubling formula
+
+As a fun application of the Bohr-Mollerup theorem, we prove the Gamma-function doubling formula
+(for positive real `s`). The idea is that `2 ^ s * Gamma (s / 2) * Gamma (s / 2 + 1 / 2)` is
+log-convex and satisfies the Gamma functional equation, so it must actually be a constant
+multiple of `Gamma`, and we can compute the constant by specialising at `s = 1`. -/
+
+/-- Auxiliary definition for the doubling formula (we'll show this is equal to `Gamma s`) -/
+def doubling_Gamma (s : ℝ) : ℝ := Gamma (s / 2) * Gamma (s / 2 + 1 / 2) * 2 ^ (s - 1) / sqrt π
+
+lemma doubling_Gamma_add_one (s : ℝ) (hs : s ≠ 0) :
+  doubling_Gamma (s + 1) = s * doubling_Gamma s :=
+begin
+  dsimp only [doubling_Gamma],
+  rw [add_div, add_assoc, add_halves (1 : ℝ), (by abel : s + 1 - 1 = s - 1 + 1),
+    Gamma_add_one (div_ne_zero hs two_ne_zero), rpow_add two_pos, rpow_one],
+  ring,
+end
+
+lemma doubling_Gamma_one : doubling_Gamma 1 = 1 :=
+by simp_rw [doubling_Gamma, Gamma_one_half_eq, add_halves (1 : ℝ), sub_self, Gamma_one, mul_one,
+  rpow_zero, mul_one, div_self (sqrt_ne_zero'.mpr pi_pos)]
+
+lemma log_doubling_Gamma_eq {s : ℝ} (hs : 0 < s) :
+  (log ∘ doubling_Gamma) s =
+  log (Gamma (s / 2)) + log (Gamma (s / 2 + 1 / 2)) + s * log 2 - log (2 * sqrt π) :=
+begin
+  have h1 : sqrt π ≠ 0, from sqrt_ne_zero'.mpr pi_pos,
+  have h2 : Gamma (s / 2) ≠ 0, from (Gamma_pos_of_pos $ div_pos hs two_pos).ne',
+  have h3 : Gamma (s / 2 + 1 / 2) ≠ 0, from (Gamma_pos_of_pos $ by positivity).ne',
+  have h4 : (2 : ℝ) ^ (s - 1) ≠ 0, from (rpow_pos_of_pos two_pos _).ne',
+  rw [function.comp_app, doubling_Gamma, log_div (mul_ne_zero (mul_ne_zero h2 h3) h4) h1,
+    log_mul (mul_ne_zero h2 h3) h4, log_mul h2 h3, log_rpow two_pos, log_mul two_ne_zero h1],
+  ring,
+end
+
+lemma doubling_Gamma_log_convex_Ioi : convex_on ℝ (Ioi (0:ℝ)) (log ∘ doubling_Gamma) :=
+begin
+  rw convex_on_iff_forall_pos,
+  refine ⟨convex_Ioi 0, λ x hx y hy a b ha hb hab, _⟩,
```

Comment:
Can't you use a more principled proof, without any computation, that a sum of convex functions is convex, that the composition of a convex function with an affine map is convex, and that a linear map is convex?

Reply:
There's really very little API for convex functions at the moment. I'll cook something up.

Reply:
There was more convex-function API than I thought, but still I found something new to add :-). I've put the additional lemmas into the `bohr_mollerup` file with a porting note to move them elsewhere.